source/common/http/websocket/ws_handler_impl.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/http/codec.h"
+#include "envoy/http/header_map.h"
+#include "envoy/http/websocket.h"
+#include "envoy/network/filter.h"
+#include "envoy/router/router.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/filter/tcp_proxy.h"
+
+namespace Envoy {
+namespace Http {
+namespace WebSocket {
+
+/**
+ * An implementation of a WebSocket proxy based on TCP proxy. This will be used for
+ * handling client connection only after a WebSocket upgrade request succeeds
+ * (i.e, it is requested by client and allowed by config). This implementation will
+ * instantiate a new outgoing TCP connection for the configured upstream cluster.
+ * All data will be proxied back and forth between the two connections, without any
+ * knowledge of the underlying WebSocket protocol.
+ */
+class WsHandlerImpl : public Filter::TcpProxy {
+public:
+  WsHandlerImpl(Http::HeaderMap& request_headers, const Router::RouteEntry& route_entry,
```

Comment:
nit: Since you are in Http namespace, you shouldn't need `Http::` in all/most places in this file and cc file. Can you remove?